src/handlers/check_commits/force_push_range_diff.rs

```diff
@@ -0,0 +1,58 @@
+use anyhow::Context as _;
+
+use crate::config::RangeDiffConfig;
+use crate::github::GithubCompare;
+use crate::github::IssueRepository;
+use crate::github::IssuesEvent;
+use crate::handlers::Context;
+
+pub(super) async fn handle_event(
+    ctx: &Context,
+    host: &str,
+    _config: &RangeDiffConfig,
+    event: &IssuesEvent,
+    compare_after: &GithubCompare,
+) -> anyhow::Result<()> {
+    if !matches!(event.action, crate::github::IssuesAction::Synchronize) {
+        return Ok(());
+    }
+
+    let (Some(before), Some(after)) = (event.before.as_ref(), event.after.as_ref()) else {
+        tracing::warn!("synchronize event but no before or after field");
+        return Ok(());
+    };
+
+    let org = event.repository.owner();
+    let repo = event.repository.name();
+
+    let compare_before = ctx
+        .github
+        .compare(
+            &IssueRepository {
+                organization: org.to_string(),
+                repository: repo.to_string(),
+            },
+            &event.issue.base.as_ref().context("no base ref")?.git_ref,
+            &before,
+        )
+        .await
+        .context("failed to get the before compare")?;
+
+    // Does the merge_base_commits differs? No, not a force-push with rebase.
+    if compare_before.merge_base_commit.sha == compare_after.merge_base_commit.sha {
+        return Ok(());
+    }
+
+    let protocol = if host.starts_with("localhost:") {
+        "http"
+    } else {
+        "https"
+    };
+
+    // Rebase detected, post a comment to our range-diff.
+    event.issue.post_comment(&ctx.github,
+        &format!(r#"This PR was rebased! Check out the changes with our [`range-diff`]({protocol}://{host}/gh-range-diff/{org}/{repo}/{before}..{after})."#)
```

Comment:
```suggestion
        &format!(r#"This PR was rebased onto a different master commit! Check out the changes with our [`range-diff`]({protocol}://{host}/gh-range-diff/{org}/{repo}/{before}..{after})."#)
```
I often do interactive rebases just to reorder commits etc. A rebase itself doesn't mean that the merge base has changed.